x/reactions/legacy/v3/store.go

```diff
@@ -0,0 +1,54 @@
+package v3
+
+import (
+	"github.com/cosmos/cosmos-sdk/codec"
+	"github.com/cosmos/cosmos-sdk/store/prefix"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/desmos-labs/desmos/v4/x/reactions/types"
+)
+
+// MigrateStore performs in-place store migrations from v2 to v3
+// It removes all the duplicated reactions.
+func MigrateStore(ctx sdk.Context, storeKey sdk.StoreKey, cdc codec.BinaryCodec) error {
+	store := ctx.KVStore(storeKey)
+	reactionsStore := prefix.NewStore(store, types.ReactionPrefix)
+
+	iter := reactionsStore.Iterator(nil, nil)
+	defer iter.Close()
+
+	// Get duplicated reactions
+	var reactions []types.Reaction
+	var duplicatedReactions []types.Reaction
+	for ; iter.Valid(); iter.Next() {
+		var reaction types.Reaction
+		err := cdc.Unmarshal(iter.Value(), &reaction)
+		if err != nil {
+			return err
+		}
+		reactions = append(reactions, reaction)
+
+		if containsDuplicatedReaction(reactions, reaction) {
+			duplicatedReactions = append(duplicatedReactions, reaction)
+		}
```

Comment:
I believe this can be done in another way that might make the code more readable: 

```go
if contains(reactions, reaction) {
  duplicatedReactions = append(duplicatedReactions, reaction)
}

reactions = append(reactions, reacion)
```

What do you think?

Reply:
Agree with it, improved